rules/REQUEST-944-APPLICATION-ATTACK-JAVA.conf

```diff
@@ -0,0 +1,210 @@
+    "msg:'Enabling body inspection',\
+    phase:1,\
+    rev:'1',\
+    ver:'OWASP_CRS/3.1.0',\
+    pass,\
+    id:944100,\
+    tag:'paranoia-level/1',\
+    ctl:forceRequestBodyVariable=On"
+SecRequestBodyAccess On
+
+
+SecRule TX:PARANOIA_LEVEL "@lt 2" "phase:1,id:944013,nolog,pass,skipAfter:END-REQUEST-944-APPLICATION-ATTACK-JAVA"
+SecRule TX:PARANOIA_LEVEL "@lt 2" "phase:2,id:944014,nolog,pass,skipAfter:END-REQUEST-944-APPLICATION-ATTACK-JAVA"
+#
+# -= Paranoia Level 2 =- (apply only when tx.paranoia_level is sufficiently high: 2 or higher)
+#
+
+
+# [ Java deserialization vulnerability/Apache Struts (CVE-2017-9805) ]
+#
+# Generic rule to detect processbuilder or runtime calls, if any of thos is found and the same target contains
+# java. unmarshaller or base64data to trigger a potential payload execution
+# tested with https://www.exploit-db.com/exploits/42627/
+
+SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_BODY|XML|XML:/* \
+    "@rx (?:runtime|processbuilder)" \
+    "msg:'Remote Command Execution: Suspicious Java method detected',\
+    phase:2,\
+    rev:'1',\
+    ver:'OWASP_CRS/3.1.0',\
+    t:none,t:lowercase,\
+    pass,\
+    id:944200,\
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-rce',\
+    tag:'OWASP_CRS/WEB_ATTACK/COMMAND_INJECTION',\
+    tag:'WASCTC/WASC-31',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/2',\
+    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}: %{MATCHED_VAR}',\
+    severity:'NOTICE',\
+    ,t:none,t:lowercase,\
+    setvar:'tx.msg=%{rule.msg}',\
+    setvar:tx.rce_score=+%{tx.notice_anomaly_score},\
+    setvar:tx.anomaly_score=+%{tx.notice_anomaly_score},\
+    setvar:tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/RCE-%{matched_var_name}=%{tx.0},\
+    setvar:tx.suspicious_method_detected=1"
+
+SecRule &tx:suspicious_method_detected "!@eq 0" \
+    "msg:'Remote Command Execution: Java process spawn (CVE-2017-9805)',\
+    phase:2,\
+    rev:'1',\
+    ver:'OWASP_CRS/3.1.0',\
+    t:none,t:lowercase,\
+    block,\
+    id:944205,\
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-rce',\
+    tag:'OWASP_CRS/WEB_ATTACK/COMMAND_INJECTION',\
+    tag:'WASCTC/WASC-31',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/2',\
+    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}: %{MATCHED_VAR}',\
+    severity:'CRITICAL',\
+    ,t:none,t:lowercase,\
+    setvar:'tx.msg=%{rule.msg}',\
+    setvar:tx.rce_score=+%{tx.critical_anomaly_score},\
+    setvar:tx.anomaly_score=+%{tx.critical_anomaly_score},\
+    setvar:tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/RCE-%{matched_var_name}=%{tx.0},\
+    chain"
+        SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_BODY|XML|XML:/* \
+            "@rx (?:unmarshaller|base64data|java\.)" \
```

Comment:
dangling and incomplete rule?

Reply:
there is an extra \ at the end, bad copy paste I guess